recurring_contract/model/contract_group.py

```diff
@@ -125,194 +112,162 @@ def write(self, cr, uid, ids, vals, context=None):
         """
         res = True
         # to solve "NotImplementedError: Iteration is not allowed" error
-        if isinstance(ids, (int, long)):
-            ids = [ids]
-
         # Any of these modifications implies generate and validate invoices
         generate_again = ('advance_billing_months' in vals or
                           'recurring_value' in vals or
                           'recurring_unit' in vals)
 
-        for group in self.browse(cr, uid, ids, context):
+        for group in self:
 
             # Check if group has an next_invoice_date
             if not group.next_invoice_date:
-                res = super(contract_group, self).write(
-                    cr, uid, group.id, vals, context) and res
+                res = super(contract_group, self).write(vals) and res
                 break
 
             # Get the method to apply changes
             change_method = vals.get('change_method', group.change_method)
             change_method = getattr(self, change_method)
 
-            res = super(contract_group, self).write(
-                cr, uid, group.id, vals, context) & res
+            res = super(contract_group, self).write(vals) & res
 
             if generate_again:
-                change_method(cr, uid, group, context)
+                change_method()
 
         if generate_again:
-            invoicer_id = self.generate_invoices(cr, uid, ids,
-                                                 context=context)
-            self.validate_invoices(cr, uid, invoicer_id, context)
+            invoicer_id = self.generate_invoices()
+            self.validate_invoices(invoicer_id)
 
         return res
-
-    def button_generate_invoices(self, cr, uid, ids, context=None):
-        invoicer_id = self.generate_invoices(cr, uid, ids, context=context)
-        self.validate_invoices(cr, uid, invoicer_id, context)
+
+    def button_generate_invoices(self):
+        invoicer_id = self.generate_invoices()

```

Comment:
Maybe we should rename `invoicer_id` to `invoicer`, as if I understand correctly, the returned value is a Record, not just an id.